uitools/cpp/Esri/ArcGISRuntime/Toolkit/NorthArrowController.cpp

```diff
@@ -20,130 +20,130 @@
-    return mapView->mapRotation();
-  else if (auto sceneView = qobject_cast<SceneView*>(m_geoView))
-    return sceneView->currentViewpointCamera().heading();
-
-  return static_cast<double>(NAN);
-}
-
-/*!
-  \brief Returns the \c GeoView as a \c QObject.
- */
-QObject* NorthArrowController::geoView() const
-{
-  return m_geoView;
-}
-
-/*!
-  \brief Set the \c GeoView object this Controller uses.
-  
-  Internally this is cast to a \c MapView or SceneView using \c qobject_cast,
-  which is why the paremeter is of form \c QObject and not \c GeoView.
-  
-  \list
-  \li \a geoView Object which must inherit from \c GeoView* and \c QObject*. 
-  \endlist
- */
-void NorthArrowController::setGeoView(QObject* geoView)
-{
-  if (geoView == m_geoView)
-    return;
-
-  disconnect(this, nullptr, m_geoView, nullptr);
-
-  m_geoView = geoView;
-
-  if (auto mapView = qobject_cast<MapViewToolkit*>(m_geoView))
+
+  /*!
+    \brief Set the heading by rotating the \c MapView or \c SceneView camera
+    to the given \a heading (in degrees).
+   */
+  void NorthArrowController::setHeading(double heading)
   {
-    connect(mapView, &MapViewToolkit::mapRotationChanged,
-            this, &NorthArrowController::headingChanged);
+    if (auto mapView = qobject_cast<MapView*>(m_geoView))
+    {
+      mapView->setViewpointRotation(heading);
+    }
+    else if (auto sceneView = qobject_cast<SceneView*>(m_geoView))
+    {
+      Camera currentCamera = sceneView->currentViewpointCamera();
+      Camera updatedCamera = currentCamera.rotateTo(heading, currentCamera.pitch(), currentCamera.roll());
+      sceneView->setViewpointCamera(updatedCamera, 0.50);
+    }
+  }
+  /*!
+    \brief Returns the calculated heading of this controller in degrees.
+   */
+  double NorthArrowController::heading() const
+  {
+    if (auto mapView = qobject_cast<MapView*>(m_geoView))
+      return mapView->mapRotation();
+    else if (auto sceneView = qobject_cast<SceneView*>(m_geoView))
+      return sceneView->currentViewpointCamera().heading();
+
+    return static_cast<double>(NAN);
   }
-  else if (auto sceneView = qobject_cast<SceneViewToolkit*>(m_geoView))
+
+  /*!
+    \brief Returns the \c GeoView as a \c QObject.
+   */
+  QObject* NorthArrowController::geoView() const
   {
-    connect(sceneView, &SceneViewToolkit::viewpointChanged,
-            this, &NorthArrowController::headingChanged);
+    return m_geoView;
   }
 
-  emit geoViewChanged();
-}
+  /*!
+    \brief Set the \c GeoView object this Controller uses.
+
+    Internally this is cast to a \c MapView or SceneView using \c qobject_cast,
+    which is why the paremeter is of form \c QObject and not \c GeoView.
+
+    \list
+      \li \a geoView Object which must inherit from \c GeoView* and \c QObject*.
+    \endlist
+  */
+  void NorthArrowController::setGeoView(QObject* geoView)
+  {
+    if (geoView == m_geoView)
+      return;
+
+    if (m_geoView)
+    {
+      disconnect(m_geoView, nullptr, this, nullptr);
+    }
```

Comment:
Bug seemed to be here. It read as:

```
disconnect(this, nullptr, m_geoView, nullptr);
```

Which on first call would do:


```
disconnect(this, nullptr, nullptr, nullptr);
```

Which is a wildcard for: remove all signals connected to `NorthArrowController` slots....

Reply:
nice found of this bug